### PR TITLE
Ruby: separate constant propagation of regexps from strings

### PR DESCRIPTION
**Pull request by nickrolfe**
This lets us distinguish constant regexp values from regular strings, which will be useful for a query I'm working on. It's similar to the way that symbols are already distinguished from strings.

As part of the change, I've deprecated the names `getStringOrSymbol` and `isStringOrSymbol`, and replaced them with `getStringlikeValue` and `isStringlikeValue`.

N.B. I checked that there were no changes to `ValueText.expected` besides the new column.

### CODE REVIEW

**Comment by nickrolfe** [comment]
This needs some work, for example to track the regexp flag string alongside its value. I'll update the PR once I've validated the approach by finishing the query I'm working on.

File: ruby/ql/test/library-tests/ast/escape_sequences/EscapeSequences.ql [hunk 1]
```diff
@@ -21,4 +21,6 @@ query predicate stringlikeLiterals(StringlikeLiteral l, string value, string kin
   value = l.getConstantValue().getString() and kind = "string"
```
Comment (hvitved) [inline ruby/ql/test/library-tests/ast/escape_sequences/EscapeSequences.ql:21]: Can now be simplified to
```ql
exists(ConstantValue v | v = l.getValueType() and value = v.getString() and kind = v.getValueType())
```

**Review by hvitved — CHANGES_REQUESTED** [review]
Looks great! We should run a DCA job before merging.

File: ruby/ql/test/library-tests/ast/escape_sequences/EscapeSequences.ql (hunk 1 above)
Comment (nickrolfe) [reply ruby/ql/test/library-tests/ast/escape_sequences/EscapeSequences.ql:21]: You mean
```ql
exists(ConstantValue v |
  v = l.getConstantValue() and value = v.getStringlikeValue() and kind = v.getValueType()
)
```
But yes, good point.

Comment (hvitved) [reply ruby/ql/test/library-tests/ast/escape_sequences/EscapeSequences.ql]: Yes, indeed.

**Comment by nickrolfe** [comment]
I merged in `main` to resolve the conflict with renamed `ParseRegExp.qll` (git was able to merge it cleanly).

**Comment by nickrolfe** [comment]
DCA performance looks fine to me.

**Pull request merged**